dashboard/src/components/dialog/dialog.tsx

```diff
@@ -0,0 +1,167 @@
+import { useAppSelector } from '@/hooks/redux.hook';
+import {
+    DialogTypeEnum,
+    selectDialog,
+    dialogActions,
+} from '@/redux/reducers/dialog.reducer';
+import React, { useState, useEffect, useRef } from 'react';
+import { useDispatch } from 'react-redux';
+
+const Dialog = () => {
+    const dialog = useAppSelector(selectDialog);
+    const dispatch = useDispatch();
+    let promptInputValue = '';
+
+    const {
+        type,
+        title,
+        message,
+        checkedMessage,
+        buttonClassName,
+        callback,
+        promptInvalidMessage,
+        isOpen,
+    } = dialog;
+
+    const [isValid, setIsValid] = useState(true);
+    const [buttonAvaible, setButtonAvaible] = useState(false);
+
+    const promptInputRef = useRef<HTMLInputElement>(null);
+    const cardRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        if (isOpen && type === DialogTypeEnum.PROMPT) {
+            promptInputRef.current?.focus();
+        } else if (isOpen) {
+            console.log('focus');
+            cardRef.current?.focus();
+        }
+        setButtonAvaible(
+            type === DialogTypeEnum.CONFIRM || type === DialogTypeEnum.ALERT
+        );
+    }, [isOpen]);
+
+    const buttonBehavior = () => {
```

Comment:
目前單看命名比較容易以為是常數，也不確定會是 canfirm or cancel btn，因此會建議：`動詞開頭 + 能辨識觸發 confirm 行為` 的命名，例如：

```suggestion
    const triggerConfirmBtn = () => {
```
不一定要這個只是舉例，但希望滿足上面的定義。

Reply:
我把它改成 do 喔

Reply:
收到